src/include/dhandle.i

```diff
@@ -0,0 +1,33 @@
+/*-
+ * Copyright (c) 2014-2018 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * __wt_dhandle_incr_use --
+ *	Increment the in-use counter in the data source.
+ */
+static inline void
+__wt_dhandle_incr_use(WT_DATA_HANDLE *dhandle)
```

Comment:
@bvpvamsikrishna you are right - we need to increment `session_inuse` not `session_ref` - since `session_ref` doesn't stop sweep from closing a handle. Could you remove `__wt_cursor_dhandle_incr_use` and call this function instead? I don't like having that functionality duplicated.